misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/evaluator/BaseFunctionTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.semver.checker.evaluator;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.ballerina.projects.Package;
+import io.ballerina.projects.directory.BuildProject;
+import io.ballerina.semver.checker.comparator.PackageComparator;
+import io.ballerina.semver.checker.diff.PackageDiff;
+import org.ballerinalang.semver.checker.ProjectUtil;
+import org.testng.Assert;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Base class for testing all function definition related test cases.
+ */
+public class BaseFunctionTest {
+
+    /**
+     * @param testFileName Name of the testcase file name
+     */
+    protected static void testEvaluate(String testFileName) throws Exception {
+        Object obj;
+        JsonArray fileData = null;
+        try (FileReader reader = new FileReader(testFileName)) {
+            obj = JsonParser.parseReader(reader);
+            fileData = (JsonArray) obj;
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
We dont need to print the stack trace in here. Lets throw an exception instead. Also we can create a `SemverTestException` class extending the `exception` class so that it'll we easier to detect any unhandled exceptions within the semver test suite